File: codeforces/98.park_lighting_[1358A]/park_lighting.cpp
#include <iostream>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int t, n, m;
  unsigned int l;
  cin >> t;
  while (t--) {
    cin >> n >> m;
    l = n * m;
    if ((l % 2) != 0) {
      l = (l + 1) / 2;
    }
    else {
      l = l / 2;
    }
    cout << l << endl;
  }
  return 0;
}

File: codeforces/low_difficulty/difficulty_800/98.park_lighting_[1358A]/park_lighting.py
# Author: José Rodolfo (jric2002)
t = int(input())
while (t):
  n, m = map(int, input().split(" "))
  l = n * m
  if ((l % 2) != 0):
    l = int((l + 1) / 2)
  else:
    l = int(l / 2)
  print(l)
  t -= 1